frontend/online-shop/src/components/ProductDetailedView.tsx

```diff
@@ -0,0 +1,123 @@
+import { Typography, Button } from "@material-ui/core";
+import Grid from "@material-ui/core/Grid/Grid";
+import { useParams, Link } from "react-router-dom";
+import { useProductDetailViewStyles } from "../styles/ProductDetailViewStyle";
+
+let theProducts = [
+  {
+    id: 1,
+    name: "Blue Jeans",
+    category: "Clothes",
+    price: 40,
+    imageUrl:
+      "https://image.made-in-china.com/2f0j00oUYGsJzBEAkl/New-Style-Casual-Fashionable-Trousers-Jeans-Men-Slim-Fit-Trousers.jpg",
+    description: "Cool stuff",
+  },
+  {
+    id: 2,
+    name: "Silver Necklace",
+    category: "Jewelery",
+    price: 60,
+    imageUrl:
+      "https://cdn.notonthehighstreet.com/fs/47/7d/2750-b4b1-412f-adba-fbfa1dbfc3b3/original_men-s-silver-or-gold-octagon-st-christopher-necklace.jpg",
+    description: "Cool stuff",
+  },
+  {
+    id: 3,
+    name: "Samsung Smart TV",
+    category: "TVs",
+    price: 500,
+    imageUrl:
+      "https://img.global.news.samsung.com/global/wp-content/uploads/2017/08/Smart-TV-Shazam_main-1_f.jpg",
+    description: "Cool stuff",
+  },
+  {
+    id: 4,
+    name: "Black T-Shirt",
+    category: "Clothes",
+    price: 20,
+    imageUrl:
+      "https://media.istockphoto.com/photos/stylish-young-woman-in-blank-black-tshirt-on-white-picture-id926335278?k=6&m=926335278&s=170667a&w=0&h=eU9hk0KC8Lst6yqQbCyzWJXoBmb9J04XnrohTjsAbj0=",
+    description: "Cool stuff",
+  },
+  {
+    id: 5,
+    name: "Aviator Sunglasses",
+    category: "Accessories",
+    price: 55,
+    imageUrl:
+      "https://www.brandfield.com/media/catalog/product/cache/image/9df78eab33525d08d6e5fb8d27136e95/8/0/805289307662.jpg",
+    description: "Cool stuff",
+  },
+  {
+    id: 6,
+    name: "Nike Running Shoes",
+    category: "Sport Shoes",
+    price: 85,
+    imageUrl:
+      "https://i.pinimg.com/originals/ef/c9/5d/efc95dfdc4acdc949274d0f7dcabd107.jpg",
+    description: "Cool stuff",
+  },
+];
+
+interface ProductProps {
+  id: number;
+  name: string;
+  category: string;
+  price: number;
+  imageUrl: string;
+  description: string;
+}
+
+function getProduct(id: number): ProductProps | undefined {
+  return theProducts.find((x) => x.id === id);
+}
+
+function ProductDetailedView() {
+  const classes = useProductDetailViewStyles();
+  let product;
```

Comment:
You could do something here like `const { id } = useParams<{ id?: string | undefined }>();  const product = !!id ? getProduct(parseInt(id)) : undefined;`.

Reply:
Also you should use a `useState` for holding the products in a stateful way. This way at the axios chapter you will make the backend call within a `useEffect` and update the products state within it.